org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/jdt/JavaDebugServer.java

```diff
@@ -23,74 +23,72 @@
 import org.eclipse.jdt.ls.debug.internal.Logger;
 
 public class JavaDebugServer implements IDebugServer {
+    private static JavaDebugServer singletonInstance;
+
     private ServerSocket serverSocket = null;
-    private Socket connection = null;
-    private ProtocolServer protocolServer = null;
+    private boolean isStarted = false;
 
     /**
-     * Constructs a JavaDebugServer instance which will launch a ServerSocket to 
-     * listen for incoming socket connection.
+     * Gets the single instance of JavaDebugServer.
+     * @return the JavaDebugServer instance
      */
-    public JavaDebugServer() {
+    public synchronized static IDebugServer getInstance() {
+        if (singletonInstance == null) {
+            singletonInstance = new JavaDebugServer();
+        }
+        return singletonInstance;
+    }
+
+    private JavaDebugServer() {
         try {
             this.serverSocket = new ServerSocket(0, 1);
         } catch (IOException e) {
-            Logger.logException("Create ServerSocket exception", e);
+            Logger.logException("Failed to create Internal Debug Server", e);
```

Comment:
There is no inter debug server now. The message should be "Failed to create the debug server."